main/src/main/scala/sbt/Load.scala

```diff
@@ -933,7 +973,11 @@ object Load {
   val LoadBuildConfiguration = sbt.LoadBuildConfiguration
 
   final class EvaluatedConfigurations(val eval: Eval, val settings: Seq[Setting[_]])
-  final case class InjectSettings(global: Seq[Setting[_]], project: Seq[Setting[_]], projectLoaded: ClassLoader => Seq[Setting[_]])
+  final case class InjectSettings(global: Seq[Setting[_]], project: Seq[Setting[_]], projectLoaded: ClassLoader => Seq[Setting[_]]) {
+    private val cache: mutable.Map[Unit, Seq[Setting[_]]] = mutable.Map.empty
+    def cachedProjectLoaded(cl: ClassLoader): Seq[Setting[_]] =
+      cache.getOrElseUpdate((), projectLoaded(cl))
```

Comment:
What @Duhemm said above: this doesn't take the input (ClassLoader) into account for subsequent calls. Is this not a problem?